task-forage.lic

```diff
@@ -0,0 +1,621 @@
+    DRCT.walk_to(target_rooms.first)
+    failures = 0
+    target_rooms.each do |room|
+      rooms[room].each do |movement|
+        return true if check_giver
+
+        old_room = Room.current.id
+        move(movement)
+        if old_room == Room.current.id
+          failures += 1
+          echo "Failure to move - #{failures}" if @debug
+        end
+        return false if failures >= 3
+      end
+    end
+    false
+  end
+
+  def get_task
+    echo 'get_task' if @debug
+    result = DRC.bput("ask #{@task_giver} for task", /I need (\d+) ([^.]+)/, /You are already on a task/,
+                      /I am sorry, you must wait/, /I have a small item that needs/, /To whom are you speaking?/)
+    case result
+    when /I need (\d+) ([^.]+)/
+      @number = Regexp.last_match(1).to_i
+      @item = Regexp.last_match(2)
+      echo "Gathering #{@number} of #{@item}" if @debug
+      return false unless evaluate_item
+
+      accept_task
+      true
+
+    when /To whom are you speaking?/
+      echo 'Task giver wandered away' if @debug
+      find_giver
+      false
+    when /I have a small item that needs/
+      echo 'Wrong task type' if @debug
+      decline_task
+      kill_time(2)
+      false
+    when /I am sorry, you must wait/
+      kill_time(1)
+      false
+    when /You are already on a task/
+      echo 'Already on a task. Canceling it' if @debug
+      cancel_task
+      kill_time(2)
+      false
+    end
+  end
+
+  def accept_task
+    echo 'accept_task' if @debug
+    case DRC.bput('accept task', 'Remember, I need', 'You have no offers to')
+    when /You have no offers to/
+      DRC.message('Task is missing.')
+      get_task
+    when /Remember, I need/
+      echo 'Task received' if @debug
+      DRCI.stow_hands
+    end
+  end
+
+  def decline_task
+    echo 'decline_task' if @debug
+    DRC.bput('decline task', 'I guess you do not wish to help', 'Please rephrase that command')
+  end
+
+  def cancel_task
+    echo 'cancel_task' if @debug
+    case DRC.bput("ask #{@task_giver} for task cancel", 'wish to end your current', 'do not currently have a',
+                  'frowns and shakes', 'To whom are you speaking?')
+    when /do not currently have a/, /frowns and shakes/
+      echo 'No task' if @debug
+      true
+    when /To whom are you speaking?/
+      find_giver
+      cancel_task
+    when /wish to end your current/
+      echo 'Cancel step 1' if @debug
+      case DRC.bput("ask #{@task_giver} for task cancel", 'I see that you no longer wish to assist',
+                    'do not currently have a')
+      when /do not currently have a/, /I see that you no longer wish to assist/
+        echo 'Cancel step 2' if @debug
+        true
+      end
+    end
+    kill_time(2) if determine_task_necessary
+  end
+
+  def use_boosts
+    echo 'use_boosts' if @debug
+    fput('info') # Check to see if a boost is already active
+    pause 0.25
+
+    lines = reget(50).reverse
```

Comment:
Could this just be a case statement rather than a reget? activated, not enough credits, or already boosted?

Reply:
Do you mean case + DRC.bput?
If so, the reason I didn't go that route is that there's no indication to match on if you don't have an active boost. The boost information (assuming there is any) is the last part that shows up in the INFO list. So if there is no boost, it'll hang until the bput 15 sec timeout

Reply:
Fair. Could just suppress the failed match and drop the timer with the bput, but understand the reasoning. I'm just not a fan of reget as it can cause problems if the room is busy. Tried using it in smoke and it kept clipping the wrong info and catching less than what I wanted.

Reply:
Oh cool, I wasn't really familiar with the bput options, but I agree that's a cleaner way to do it.